vignettes/flowmaps-static.qmd

```diff
@@ -0,0 +1,774 @@
+```
+
+## Aggregate data - count total flows
+
+```{r}
+od_20210407_total <- od_20210407 |>
+  group_by(id_origin, id_destination) |>
+  summarise(n_trips = sum(n_trips, na.rm = TRUE)) |> 
+  collect()
+
+head(od_20210407_total)
+```
+
+```
+# A tibble: 6 × 3
+# Groups:   id_origin [2]
+  id_origin id_destination  n_trips
+  <fct>     <fct>             <dbl>
+1 0704003   07031           3958.  
+2 0704003   07036          17278.  
+3 0704003   07044            572.  
+4 0704003   0801903            6.19
+5 0704004   07027           3036.  
+6 0704004   0704003        56604.
+```
+
+## Reshape flows for visualization
+
+We are going to use the `{flowmapper}` [@flowmapper-r] package to plot the flows. This package expects the data to be in the following format:
+
+```{r}
+'data.frame':	7 obs. of  8 variables:
+ $ id_a   : chr  "X1" "X2" "X3" "X3" ...
+ $ id_b   : chr  "X5" "X4" "X1" "X5" ...
+ $ xa     : num  2 14 10 10 2 14 14
+ $ ya     : num  6 10 9 9 6 10 10
+ $ xb     : num  10 4 2 10 4 10 10
+ $ yb     : num  4 10 6 4 10 4 9
+ $ flow_ab: num  1 2 3 3 1 1 4
+ $ flow_ba: num  2 3 2 5 2 1 5
+```
+
+### Reshape the flows
+
+The data we have right now in `od_20210407_total` is in a classic long format. That is, we have rows for both the number of trips from A to B and from B to A. `{flowmapper}` requires the data to be in different format, where there is only one row for each pair of `id_a` and `id_b` and two columns, one for the flow from `id_a` to `id_b` and one for the flow from `id_b` to `id_a`.
+
+
+```{r}
+od_20210407_total_twoway <- od_20210407_total |>
+  mutate(id_origin = as.character(id_origin),
+    id_destination = as.character(id_destination)
+  ) |>
+  # Rename columns to match the desired output
+  rename(id_a = id_origin, id_b = id_destination, flow_ab = n_trips) |>
+  # Create the reverse flow
+  full_join(
+    od_20210407_total |>
+      rename(id_b = id_origin, id_a = id_destination, flow_ba = n_trips),
+    by = c("id_a", "id_b")
+  ) |>
+  # Replace NA with 0 for flows that don't exist in one direction
+  mutate(
+    flow_ab = coalesce(flow_ab, 0),
+    flow_ba = coalesce(flow_ba, 0)
+  ) |>
+  # Remove duplicate rows (where id_a and id_b are swapped)
+  filter(id_a <= id_b) |>
+  # Sort by id_a and id_b
+  arrange(id_a, id_b) |> 
+  # Remove the flows within the autonomous communities
+  filter(id_a != id_b)
+
+head(od_20210407_total_twoway)
+```
+
+```
+# A tibble: 6 × 4
+# Groups:   id_a [1]
+  id_a     id_b     flow_ab flow_ba
+  <chr>    <chr>      <dbl>   <dbl>
+1 01001_AM 01002      20.2     29.6
+2 01001_AM 01010_AM   15.6      0  
+3 01001_AM 01031_AM   48.6     63.3
+4 01001_AM 01036      39.8     65.2
+5 01001_AM 01043_AM  387.     388. 
+6 01001_AM 01047_AM    4.15   109. 
+```
+
+### Add coordinates
+
+We need the coordinates for each origin and destination. We can use the centroids of `districts_v1` polygons for that.
```

Comment:
Is this something that can be done with `{od}`? I found the function `od_centroids`, but it gives weird results, now what I would expect compared to the code I've written below.

Reply:
Should work. Can check. Will be great to compare {od} with {flowmapper}.